openxr/examples/vulkan.rs

```diff
@@ -0,0 +1,649 @@
+                // Don't grind up the CPU
+                std::thread::sleep(Duration::from_millis(100));
+                continue;
+            }
+
+            let state = frame_wait.wait().unwrap();
+            frame_stream.begin().unwrap();
+
+            if !state.should_render {
+                frame_stream
+                    .end(
+                        state.predicted_display_time,
+                        xr::EnvironmentBlendMode::OPAQUE,
+                        &[],
+                    )
+                    .unwrap();
+                continue;
+            }
+
+            if swapchain.is_none() {
+                let views = instance
+                    .enumerate_view_configuration_views(
+                        system,
+                        xr::ViewConfigurationType::PRIMARY_STEREO,
+                    )
+                    .unwrap();
+                let resolution = vk::Extent2D {
+                    width: views[0].recommended_image_rect_width,
+                    height: views[0].recommended_image_rect_height,
+                };
+                let handle = session
+                    .create_swapchain(&xr::SwapchainCreateInfo {
+                        create_flags: xr::SwapchainCreateFlags::EMPTY,
+                        usage_flags: xr::SwapchainUsageFlags::COLOR_ATTACHMENT
+                            | xr::SwapchainUsageFlags::SAMPLED,
+                        format: COLOR_FORMAT.as_raw() as _,
+                        sample_count: 1,
+                        width: resolution.width,
+                        height: resolution.height,
+                        face_count: 1,
+                        array_size: VIEW_COUNT,
+                        mip_count: 1,
+                    })
+                    .unwrap();
+                let images = handle.enumerate_images().unwrap();
+                swapchain = Some(Swapchain {
+                    handle,
+                    resolution,
+                    buffers: images
+                        .into_iter()
+                        .map(|color_image| {
+                            let color_image = vk::Image::from_raw(color_image);
+                            let color = vk_device
+                                .create_image_view(
+                                    &vk::ImageViewCreateInfo::builder()
+                                        .image(color_image)
+                                        .view_type(vk::ImageViewType::TYPE_2D)
+                                        .format(COLOR_FORMAT)
+                                        .subresource_range(vk::ImageSubresourceRange {
+                                            aspect_mask: vk::ImageAspectFlags::COLOR,
+                                            base_mip_level: 0,
+                                            level_count: 1,
+                                            base_array_layer: 0,
+                                            layer_count: 2,
```

Comment:
2 for left and right viewports, I assume?

Reply:
Made this a little more explicit.